packages/events/defaultSubscribers/dashboard.js

```diff
@@ -39,6 +39,20 @@ module.exports = {
         }
       }
     ],
+    "compile:succeed": [
+      async function ({ result }) {
+        try {
+          const publishLifecycle = await this.messageBusClient.publish({
+            type: "cli-event",
```

Comment:
What's the value of a generic `cli-event`, vs. making a dedicated workflow-compile-result message type? Do we really want to shoehorn a bunch of different events like this? I know Ben was talking about making things generic to avoid the use of `any`, but we could also... just make message types for message types we expect.

Reply:
I'm torn, Ben's most convincing comment imo is [this one](https://github.com/trufflesuite/truffle/pull/5621#discussion_r1020108728). Maybe we can reconsider this one when dashboard starts receiving CLI events of more types?

As an aside, I just thought about the plugin system and it will probably require a generic message type similar to this one.

Reply:
FWIW, my biggest (albeit not my only) concern about the specific message type was that it was causing us to take a dependency on `workflow-compile` from `dashboard-message-bus-common`.

My thought there is that over time as the scope of the dashboard grows, it's likely going to require data from a bunch of different packages. My fear is that if we establish a pattern of coupling generic/core/common packages to specific/functional packages we're likely going to run into a bunch of obnoxious problems (e.g. circular dependencies) down the line. Admittedly in this case the logic wasn't tightly coupled, but in my view that's more reason to avoid taking the direct dependency.

As a secondary concern, I think it's also usually a bad sign for the maintainability of an architecture when "transport" systems need to be made aware of the detailed structure of the data that it's passing. Ideally we'd have a single envelope type for all messages, there would be some topic key in that type, and the bus would just route messages to relevant subscribers based on that topic key. The payload field of that message envelope would also ideally be typed `unknown` so that we'd get compiler errors if anything w/i the message bus logic tried to interact with it directly.